tests/base_test.go

```diff
@@ -0,0 +1,606 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/go-logr/logr"
+	apis "github.com/konveyor/forklift-controller/pkg/apis"
+	"github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1"
+	"github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1/plan"
+	"github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1/provider"
+	"github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1/ref"
+	"github.com/konveyor/forklift-controller/pkg/controller/provider/container/ovirt"
+	"github.com/konveyor/forklift-controller/pkg/settings"
+	ovirtsdk "github.com/ovirt/go-ovirt"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+func TestSanityOvirtProvider(t *testing.T) {
+	var Settings = &settings.Settings
+	var log logr.Logger
+
+	conf, err := config.GetConfig()
+	if err != nil {
+		fmt.Println("unable to set up client config")
+		os.Exit(1)
+	}
+
+	logf.SetLogger(
+		logf.ZapLogger(Settings.Logging.Development))
+	log = logf.Log.WithName("entrypoint")
+
+	username := os.Getenv("OVIRT_USERNAME")
+	if username == "" {
+		t.Fatal("OVIRT_USERNAME is not set")
+	}
+	password := os.Getenv("OVIRT_PASSWORD")
+	if password == "" {
+		t.Fatal("OVIRT_PASSWORD is not set")
+	}
+
+	ovirtURL := os.Getenv("OVIRT_URL")
+	if ovirtURL == "" {
+		t.Fatal("OVIRT_URL is not set")
+	}
+
+	cacertFile := os.Getenv("OVIRT_CACERT")
+	if cacertFile == "" {
+		t.Fatal("OVIRT_CACERT is not set")
+	}
+
+	cacert, err := os.ReadFile(cacertFile)
+	if err != nil {
+		t.Fatalf("Could not read %s", cacertFile)
+	}
+
+	sc := os.Getenv("STORAGE_CLASS")
+	if sc == "" {
+		t.Fatal("STORAGE_CLASS is not set")
+	}
+
+	vmId := os.Getenv("OVIRT_VM")
```

Comment:
yes, i wrote it in the commit itself in a TODO. also to clean the code, and maybe check other validations (on the plan status + check the vm existence in the namespace).

Reply:
> Why is it better to get a template? In any case something needs to be there already?

either a template or an image - it would save the time to install the vm (by using a pre-installed image in both cases)